public/app/features/query/components/QueryEditorRow.tsx

```diff
@@ -567,7 +567,11 @@ export function filterPanelDataToQuery(data: PanelData, refId: string): PanelDat
 
   // Only say this is an error if the error links to the query
   let state = data.state;
-  const error = data.error && data.error.refId === refId ? data.error : undefined;
+  let error = data.errors?.find((e) => e.refId === refId);
```

Comment:
can there be multiple for the same refId?  should this be filter?

Reply:
Multiple errors per ref id shouldn't be possible given contract define a data response per ref id and it can only hold one error https://github.com/grafana/grafana-plugin-sdk-go/blob/3afcd6cc62b6203fe93bfd70a0f7ecf3089fcee2/proto/backend.proto#L131.